### PR TITLE
Feature: Space group symmetry for EXX based on LibRI v0.2.1.0

### LINKED ISSUE #2881: Symmetry for EXX
## Background
Currently, EXX calculation does not support symmetry=0 and 1, due to wrong DM(R) (for all the functionals) and wrong Vs (for HF and PBE0). If symmetry is supported, the PBE part calculation can be reduced to a rate of nks_ibz/nks_tot.
## To Implement
- symmetry=0, DM(R) and Vs (#2868)
- symmetry=1, DM(R)

### PR DESCRIPTION
**Pull request by maki49**
fix https://github.com/deepmodeling/abacus-develop/issues/2881

#### Prerequisites
- https://github.com/deepmodeling/abacus-develop/pull/4830
- https://github.com/abacusmodeling/LibRI/pull/4 

#### Features
1. Rotate DM(k) to reduce k-points in the hybrid functional process.
2. Rotate Hexx(R) to reduce the real-space sector in EXX calculation (based on LibRI v0.2.1.0).

#### Acknowledgements
- [PeizeLin](https://github.com/PeizeLin)
- [minyez](https://github.com/minyez)

### CODE REVIEW

**Comment by maki49** [comment]
I changed the previous irreducible k-points to their equivalent ones to ensure they are originally generated by MP sampling rather than their rotation. If not doing this, the symmetry operation from IBZ-k-point to the original k-point may not be a true symmetry operation of the BvK supercell, which will lead to wrong Hexx(R)s.
image
However, this change affects the results of a few test cases. I will talk to the related developers and change the reference results.

File: source/module_ri/module_exx_symmetry/irreducible_sector.cpp [hunk 1]
```diff
@@ -0,0 +1,238 @@
+#include "module_ri/module_exx_symmetry/irreducible_sector.h"
+
+namespace ModuleSymmetry
+{
```
Comment (mohanchen) [inline source/module_ri/module_exx_symmetry/irreducible_sector.cpp:4]: I would suggest adding some comments to these codes in this source file

File: source/module_ri/module_exx_symmetry/irreducible_sector_bvk.cpp [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+                            }
+                            ++at;
+                        }
+}
+}
+}
+}
+
+        // analyze bravis and generate optimized lattice for minimal BvK lattice
+        double cel_const[6], pre_const[6];
```
Comment (mohanchen) [inline source/module_ri/module_exx_symmetry/irreducible_sector_bvk.cpp]: I would suggest one variable for one line

File: source/module_ri/module_exx_symmetry/irreducible_sector_bvk.cpp [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        int bvk_brav;
```
Comment (mohanchen) [inline source/module_ri/module_exx_symmetry/irreducible_sector_bvk.cpp]: initialize the variable

File: source/module_ri/module_exx_symmetry/irreducible_sector.cpp (hunk 1 above)
Comment (maki49) [reply source/module_ri/module_exx_symmetry/irreducible_sector.cpp:4]: There have been many comments in the corresponding header telling what each function does.

**Review by mohanchen — APPROVED** [review]

**Pull request merged**